Fixed broken Filter::XSSFilter() unit tests

Kristuff opened the pull request:
Currently, `testXSSFilterWithNonStringArguments()` method in `tests\core\FilterTest.php` does not test the  `XSSFilter()` method. Changed  from:

```php
public function testXSSFilterWithNonStringArguments()
{
   $this->assertEquals(123, 123);
   $this->assertEquals(array(1, 2, 3), array(1, 2, 3));
   $this->assertEquals(17.001, 17.001);
   $this->assertEquals(null, null);
}
```
to:
```php
public function testXSSFilterWithNonStringArguments()
{
    $this->assertEquals(123, Filter::XSSFilter(123));
    $this->assertEquals(array(1, 2, 3), Filter::XSSFilter(array(1, 2, 3)));
    $this->assertEquals(17.001, Filter::XSSFilter(17.001));
    $this->assertEquals(null, Filter::XSSFilter(null));
}
```

panique commented:
Excellent man, excellent It's so weird that these tests did basically nothing :)

Kristuff commented:
Yep ;) But unfortunately, the test does not pass. I will investigate.

panique commented:
Something interesting happened: Every time somebody wants to merge something, the code is run by Travis, an external company that does free builds for open source software. HUGE is tested against PHP 5.6, 7.0, 7.1 and HHVM, and this merge request fails in HHVM, as HHVM doesn't know references in PHP, like `test(&$stuff)`. Wow, that's special! Any ideas how to handle this ? I'm totally clueless right now...

https://travis-ci.org/panique/huge/builds/222599422

```
Fatal error: Cannot pass parameter 1 by reference in /home/travis/build/panique/huge/tests/core/FilterTest.php on line 21
```

panique commented:
Aaah i think i found the bug :) The last test uses `null` as parameter, which will not work in this case, as null cannot be handled as a variable inside the function itself. Can you just remove that line that tests with null ? Big Thanks for the tests, i'll then merge instantly!

Kristuff commented:
For my first contribution here, sorry to inconvenience :)

Sorry, I could remove the using of `null` but it won't change anlything. I already made all test combinaisons. 

In fact, you also noted the tests also failed with PHP 5.6., and be carrefull the raison they pass with PHP7.

I am a quit new to PHPunit and travis (and even Github). After some tests in a dummy project https://github.com/Kristuff/dummyPHP, I have reproduced that error with all PHP versions (5.3 to 7.1 and hhvm) and fixed it by changing the test method like in last commit. Using of *dedicaded* method like `assertNull` instead `assertEquals($null, ...)` is optional but the *recommanded* way, it's why I changed it (the two ways work like a charm in my test project).

The second issue (still not fixed) is the PHPUnit version used on travis. 
The PHPUnit version running in my test project is 5.7.19 on PHP 5.6 when it is 5.7.15 here (I forced install and run of last *stable* PHPUnit)

And concerning PHP 7 and 7.1. I already seen the issue. Travis currently uses the default version of PHPUnit compatible with PHP7, and it's now 6.0 . Tests class are not compatibles with the new namespace synthax. The log says **'No tests executed!'** and it's why the tests pass! 

Note that it is also the case for all other tests of this project (I check another recent pull request). Currently, there is no tests executed of PHP7 at all.

I think we need to force travis using the last compatible PHPUnit version (5.7.* which is supported on PHP 5.6, 7.0, 7.1). I will try to make the change soon.

And by the way, thank you very much for this project. I learned a lot with it :)

Kristuff commented:
Great, it finally pass! :)

panique commented:
Thanks! :) I've removed the weird StyleCI syntax check as it had strange results

Kristuff commented:
Hi, 
I see an attempt to send code coverage report to Scrutinizer in `travis.yml`. 
```
# gets tools from Scrutinizer, uploads unit tests results to Scrutinizer (?)
...
```
The code coverage report is not generated for now because there is no white list defined. I have configured the white list in phpunit.xml configuration like this:

panique commented:
Thanks a lot, awesome! I'm adding this to the project

Kristuff commented:
Your are welcome. To close the XssFilter chapter (I hope :)... ), I see an open issue https://github.com/panique/huge/issues/790 and a blocked PR https://github.com/panique/huge/pull/794 with [TODO] concerning a recursive implementation of XssFilter. I have implemented and fully tested the solution proposed here https://github.com/panique/huge/issues/790  (the method accepts now `string`, `array` and `object`).

I can try to make another PR? My tests passed here https://travis-ci.org/Kristuff/huge/builds/225608242

panique commented:
Awesome!! Can you please make a PR, this would be super cool! Thanks a lot @Kristuff

Kristuff commented:
done :)
